### ISSUE TITLE
[question] How to make the message streams work for inter process communication. 

### ISSUE THREAD

**Issue by falcon027** [davedoesdev/shared-memory-disruptor#28]
_This is more of a question than a problem._

 I was trying to build an IPC pub sub implementation using this module in particular the stream functionality. In the test setup seen below it seems to me that some data is missing or not in proper order. I am not sure if this is a problem with my message framing or a lack of understanding of the LMAX disruptor. 

#### producer.js

```javascript
const {Disruptor, DisruptorWriteStream} = require('shared-memory-disruptor');

let stream = new DisruptorWriteStream(
    new Disruptor('/stream', 5000, 1, 1, 0, false, false)
)

function push(data) {
    return new Promise((resolve) => {
        stream.write(JSON.stringify(data) + '\0', function (data) {
            resolve()
        });
    })
}

async function start() {
    for (let i = 0; i < 1000; i++) {
        await push({test: "Text", data: i})
    }
}

start()
```

#### consumer.js
```javascript
const {Disruptor, DisruptorReadStream} = require('shared-memory-disruptor');

const d = new Disruptor('/stream', 5000, 1, 1, 0, true, false);
const rs = new DisruptorReadStream(d)


let msg = '';
let msgCount = 0
rs.on('data', (chunk) => {
    msg += chunk.toString('utf8');
    msgCount += 1
    if (msg.charCodeAt(msg.length - 1) === 0) {
        let allData = msg.substring(0, msg.length - 1)
        allData.split("\0").map(message => {
            try {
                let data = JSON.parse(message)
                console.log(data);
            } catch (e) {
                console.log(message);
            }
        })
        msg = '';
        msgCount = 0
    }
});


```

#### problem (output)

I would expect the output to contain all the messages sent in order, but some messages are messed up as shown below.
```
{ test: 'Text', data: 556 }
{ test: 'Text', data: 557 }
{ test: 'Text', data: 558 }
{"test":"Text","dt",ext","d{"test":"Text","data":560}
{ test: 'Text', data: 561 }
{ test: 'Text', data: 562 }
```
If anyone has an idea as to the reason for this behaviour, please get in touch with me.

Thanks in advance 
_Falcon_

**Comment by davedoesdev** [comment]
@falcon027 congrats, you found a bug!

I have a fix:

```diff
diff --git a/lib/disruptor.js b/lib/disruptor.js
index 92c497c..7bc4418 100644
--- a/lib/disruptor.js
+++ b/lib/disruptor.js
@@ -267,7 +267,7 @@ class DisruptorWriteStream extends Writable {
         }
         let i = 0;
         for (const buf of bufs) {
-            chunk.copy(buf, 0, i, buf.length);
+            chunk.copy(buf, 0, i, i + buf.length);
             i += buf.length;
         }
         if (i === 0) {
```

I'll publish it soon but I want to get your example into the tests first.

**Comment by davedoesdev** [comment]
@falcon027 fixed in version 4.1.4

**Comment by falcon027** [comment]
I appreciate the quick fix, it works now, have a good day.
